fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -27,23 +27,54 @@
  * {@link KeyBindingRegistry#register(FabricKeyBinding)}!
  */
 public class FabricKeyBinding extends KeyBinding {
-	protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String category) {
-		super("key." + id.toString().replace(':', '.'), type, code, category);
+	protected FabricKeyBinding(String name, InputUtil.Type type, int code, String category) {
+		super(name, type, code, category);
 	}
 
 	public static class Builder {
-		protected final FabricKeyBinding binding;
 
-		protected Builder(FabricKeyBinding binding) {
-			this.binding = binding;
+	    private InputUtil.Type type = InputUtil.Type.KEYSYM;
+
+		private String keyName = "key.fabric.unnamed";
+
+		private int code;
+
+		private String category = KeyCategory.MISC;
+
+		private Builder() {
+
+		}
+
+		public Builder name(String keyName) {
+		    this.keyName = keyName;
+		    return this;
+		}
+
+		public Builder name(Identifier keyName) {
+		    return name("key." + keyName.toString().replace(':', '.').replace('/', '/'));
+		}
+
+		public Builder category(String category) {
+		    this.category = "key.categories." + category;
+		    return this;
+		}
+
+		public Builder code(int keyCode) {
+		    this.code = keyCode;
+		    return this;
+		}
+
+		public Builder type(InputUtil.Type type) {
+		    this.type = type;
+		    return this;
 		}
 
 		public FabricKeyBinding build() {
-			return binding;
+			return new FabricKeyBinding(keyName, type, code, category);
 		}
 
-		public static Builder create(Identifier id, InputUtil.Type type, int code, String category) {
```

Comment:
Keep this, but mark as deprecated. Give it impl like `create().identifier(id).type(type).code(code).category(category);`

Reply:
That's what I ended up doing, though with an added change to raw categories will work.